### PR TITLE
Add EAD slurp and include into foxmlToSolr.xslt.

### PR DESCRIPTION
**Pull request by adam-vessey**

### CODE REVIEW

File: islandora_transforms/slurp_all_ead_to_solr.xslt [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- CML -->
```
Comment (willtp87) [inline islandora_transforms/slurp_all_ead_to_solr.xslt]: copypasta alert.

File: islandora_transforms/slurp_all_ead_to_solr.xslt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+<xsl:stylesheet version="1.0"
+  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+  xmlns:foxml="info:fedora/fedora-system:def/foxml#">
+  <xsl:template match="foxml:datastream[@ID='EAD']/foxml:datastreamVersion[last()]" name="index_EAD">
+    <xsl:param name="content"/>
+    <xsl:param name="prefix"></xsl:param>
+    <xsl:param name="suffix">ms</xsl:param>
+
+    <xsl:apply-templates mode="slurping_EAD" select="$content//*[local-name() = 'ead'][1]">
```
Comment (willtp87) [inline islandora_transforms/slurp_all_ead_to_solr.xslt]: Not sure there can be multiple ead elements.

File: islandora_transforms/slurp_all_ead_to_solr.xslt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  <xsl:template match="*" mode="slurping_EAD">
+    <xsl:param name="prefix"/>
+    <xsl:param name="suffix"/>
+
+    <xsl:variable name="lowercase" select="'abcdefghijklmnopqrstuvwxyz_'" />
+    <xsl:variable name="uppercase" select="'ABCDEFGHIJKLMNOPQRSTUVWXYZ '" />
+
+    <xsl:variable name="this_prefix">
+      <xsl:value-of select="concat($prefix, local-name(), '_')"/>
+      <xsl:if test="@type">
```
Comment (willtp87) [inline islandora_transforms/slurp_all_ead_to_solr.xslt:31]: There may be other attributes for example 'label' we should be sniffing into the field name.

**Comment by willtp87** [comment]
Not sure if it will be of use but I noticed an attribute on dates called 'normal' which may help in date parsing.

File: islandora_transforms/slurp_all_ead_to_solr.xslt (hunk 3 above)
Comment (adam-vessey) [inline islandora_transforms/slurp_all_ead_to_solr.xslt:31]: `@label` appears to be more for output in UI? I saw something like:

``` xml
<abstract label="Abstract">[...]</abstract>
```

**Comment by adam-vessey** [comment]
`@normal` appears to contain a normalized copy of the date, but I only found a single instance of it in the EAD doc I examined, and it was representing a range, so... Question mark?

File: islandora_transforms/slurp_all_ead_to_solr.xslt (hunk 3 above)
Comment (willtp87) [inline islandora_transforms/slurp_all_ead_to_solr.xslt:31]: That's probably the intent but there is semantic meaning in `<unitid label="Collection number">LSC.1923</unitid>` , `<physdesc label="Extent">`and `<origination label="Creator">` perhaps, but we probably shouldn't support it in the base case.  How about `level`?

Comment (adam-vessey) [inline islandora_transforms/slurp_all_ead_to_solr.xslt:31]: Doesn't seem to offer anything not already described... `c01` -> `@level` == 'series', `c02` -> `@level` == 'file'...

**Pull request merged**